test/tests/kernels/ad_value/generic_value.i

```diff
@@ -0,0 +1,81 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 2
+  ny = 2
+[]
+
+[Variables]
+  [./u]
+  [../]
+  [./v]
+  [../]
+[]
+
+[AuxVariables]
+  [./u_jac]
+  [../]
+  [./v_jac]
+  [../]
+[]
+
+[Kernels]
+  [./diff]
+    type = Diffusion
```

Comment:
Shouldn't this be ADDiffusion?

Reply:
In theory...I just copied the ad_value test and replaced the ADValueTest with the ADGenericValueTest to show the correct kernel is being called.

Reply:
@recuero, with the original test only having 1 AD class, doesn't this just default to using the non-AD value?  The ADKernel is called via the ADGenericKernel, but the variable u can't be mixed between both styles?

Reply:
You can have AD and non-AD `kernels` contributing to the residual and Jacobian simultaneously. The type for `u` within the `ADKernel` is `ADTemplateVariableValue`, so I think this is converted behind the scenes; but I'd make sure the right classes are being called.

Or better yet, you could duplicate another test object that only uses one kernel type (since kernel interactions do not need to be tested here), which would make it more straightforward. Both versions of AD and both versions of non-AD should provide identical results, same number of iterations, etc.

Reply:
There is not much conversion required. The ADKernel on residuals is not computing derivatives, so only the raw value will have to be inserted into the PETSc residual vector. For a Jacobian computation the dual number derivatives get inserted into the PETSc Jacobian matrix. PETSc does not know dual numbers.

Reply:
So, it's not converted because it already has the right type for an ADKernel